src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestDiscoveryEventsHandler2.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.Interfaces
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Interface contract for handling discovery events during test discovery operation
+    /// </summary>
+    public interface ITestDiscoveryEventsHandler2 : ITestMessageEventHandler
```

Comment:
Could have inherited from `ITestDiscoveryEventsHandler' added `void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryCompleteEventArgs, IEnumerable<TestCase> lastChunk);`

Reply:
We will be exposing this Interface to users. In inheriting from old interface case, users have to implement all the methods. So, that's why didn't go with that approach.  I have already written converter so that existing clients do not break which I will raise in next PR.